Assignment 7B/part_1/js/script.js

```diff
@@ -0,0 +1,200 @@
+        //begin step 6 
+        clearLoading(); //clears value of right div
+        //end step 6 (continued below also)
+        rightSide.appendChild(h1Element); //appending element to right div
+        rightSide.appendChild(h2Element); //appending element to right div
+        rightSide.appendChild(ulElement); //appending element to right div
+        ulElement.appendChild(li1); //appending li to ul
+        ulElement.appendChild(li2); //appending li to ul
+        ulElement.appendChild(li3); //appending li to ul
+        rightSide.appendChild(imageElement); //added img element to rightSide as part of step 5
+        fetchWeatherIcon(imageElement, xmlDoc); //added as part of step 5 -- to fetch the weather icon and add it to the div id right
+    }
+
+    function createElement(elementChoice, textChoice){ //function to create an element with a passed innerHTML text
+        let element = document.createElement(elementChoice); //creating passed element type
+        element.innerHTML = textChoice; //setting passed text value to element
+        return element; //returning element
+    }
+
+    //end step 4
+    //begin step 5
+    function fetchWeatherIcon(elementName, xmlDoc){ //function to fetch the weather icon 
+        //begin step 6
+        let siteURL = "http://forecast.weather.gov/images/wtf/large/"; //set the site url to the actual URL and changed small to large
+        //end step 6
+        let iconURL = xmlDoc.getElementsByTagName("icon_url_name")[0].innerHTML; //setting icon url via what is returned in the xmlDoc that was fetched previously
+        let completeURL = `${siteURL}${iconURL}`; //adding the two url pieces together in a template literal
+        fetch(completeURL, { //fetching icon with built in request interface
+            method: "GET", //GET method to recieve data
+            mode: "cors", //cors enabled to allow fetch outside of domain
+            redirect: "follow" //if a redirect happens -- follow it 
+        })
+          .then((response) => { //after respone is received
+              if(response.ok){ //if response is successful / 200 status
+                  console.log(response); //logging response for my own verification
+                  return response.blob(); //returning response received as a blob object
+              }
+              throw new Error(response.statusText) //if error -- throw status text
+          })
+            .then(blob => { //using blob data
+                console.log(blob); //logging blob for my own verification
+                elementName.src = URL.createObjectURL(blob); //setting the passed elements src value to return value of URL.createObjectURL (so the src knows where to look) -- to display image on the element
+            })
+             .catch(error => console.log("Error retrieving icon:     " + error)); //if error is thrown -- log to console
+    }
+    //end step 5
+
+
+
```

Comment:
Very nice! You do have it all working - and it is a bit of a beast :) I can see that you would clean some things up with more time - formatting and such - but your functionality and comments are good!